protected/components/PatientSearch.php

```diff
@@ -204,6 +205,30 @@ public function getPatientName($term)
     }
 
     /**
+     * Fetch contact name and title from search term
+     */
+    public function getContactTitleName($term)
+    {
+        $contacts_result = Contact::model()->findAll(array('select'=>'title', 'distinct'=>true,));
+        $titles = array();
+        foreach ($contacts_result as $contact) {
+            array_push($titles, strtolower($contact['title']));
+        }
+
+        $terms = explode(' ', $term, 3);
+        $result = null;
+        $result['contact_name'] = array();
+        foreach ($terms as $term){
```

Comment:
I'm not sure if it would be more readable, but you could replace this loop with something like this:
$result['title'] = implode(array_intersect($terms, $titles))
$result['contact_name'] = array_diff($terms, $titles);